error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/PrimitiveRules.java

```diff
@@ -384,4 +386,98 @@ boolean after(double d) {
       return Double.isFinite(d);
     }
   }
+
+  /** Prefer an {@link Integer#signum(int)} comparison to 1 over less clear alternatives. */
+  static final class IntegerSignumIsPositive {
+    @BeforeTemplate
+    boolean before(int i) {
+      return Refaster.anyOf(Integer.signum(i) > 0, Integer.signum(i) >= 1);
+    }
+
+    @AfterTemplate
+    @AlsoNegation
+    boolean after(int i) {
+      return Integer.signum(i) == 1;
+    }
+  }
+
+  /** Prefer an {@link Integer#signum(int)} comparison to -1 over less clear alternatives. */
+  static final class IntegerSignumIsNegative {
+    @BeforeTemplate
+    boolean before(int i) {
+      return Refaster.anyOf(Integer.signum(i) < 0, Integer.signum(i) <= -1);
+    }
+
+    @AfterTemplate
+    @AlsoNegation
+    boolean after(int i) {
+      return Integer.signum(i) == -1;
+    }
+  }
+
+  /** Prefer an {@link Long#signum(long)} comparison to 1 over less clear alternatives. */
+  static final class LongSignumIsPositive {
+    @BeforeTemplate
+    boolean before(long l) {
+      return Refaster.anyOf(Long.signum(l) > 0, Long.signum(l) >= 1);
+    }
+
+    @AfterTemplate
+    @AlsoNegation
+    boolean after(long l) {
+      return Long.signum(l) == 1;
+    }
+  }
+
+  /** Prefer an {@link Long#signum(long)} comparison to -1 over less clear alternatives. */
+  static final class LongSignumIsNegative {
+    @BeforeTemplate
+    boolean before(long l) {
+      return Refaster.anyOf(Long.signum(l) < 0, Long.signum(l) <= -1);
+    }
+
+    @AfterTemplate
+    @AlsoNegation
+    boolean after(long l) {
+      return Long.signum(l) == -1;
+    }
+  }
+
+  /** Prefer a {@link Math#signum(double)} comparison to 1 over less clear alternatives. */
+  static final class MathSignumIsPositive {
+    @BeforeTemplate
+    boolean before(float v) {
+      return Refaster.anyOf(Math.signum(v) > 0, Math.signum(v) >= 1);
+    }
+
+    @BeforeTemplate
+    boolean before(double v) {
+      return Refaster.anyOf(Math.signum(v) > 0, Math.signum(v) >= 1);
+    }
+
+    @AfterTemplate
+    @AlsoNegation
+    boolean after(double v) {
+      return Math.signum(v) == 1;
```

Comment:
Here and in the `@AfterTemplaye` below flags [_S1244: Equality tests should not be made with floating point values_](https://sonarcloud.io/organizations/picnic-technologies/rules?open=java%3AS1244&rule_key=java%3AS1244), which is fair. Options:
1. Omit the `Math.signum` rules.
2. Suppress the warning, since the this will do the right thing, as per the documentation.

Option (2) does force others to suppress the warning too, though, so perhaps (1) is better?

Reply:
Oof, returning a double/float that is fixed to `1.0 / 0.0 / -1.0` is an interesting design indeed 😿 IMO generally most uses of double/float should be considered very carefully, but more to the point, I guess I'd lean towards the consistency and keep it included.

Reply:
Hmm, IIUC the equality check *will* work right? Because the floating point of 1 equality check is just fine. However, based on the Sonar rule I'd say we should not rewrite this one. It would maybe be better to flag this one as something that is not something one should do in code. 

Additionally, for people outside of Picnic that have setup Sonar with most rules enabled will have to suppress / work around this one after our rewite, if I'm correct.

Reply:
K, I rebased and dropped the `Math` rules, at least for now. That way we can merge the PR and defer the rest (to potentially never do it).